pkg/products/marin3r/apiUsagePrometheusRules.go

```diff
@@ -41,60 +41,97 @@ func (r *Reconciler) newAlertsReconciler(grafanaConsoleURL string) (resources.Al
 // mapAlertsConfiguration maps each value from alertsConfig into a
 // resources.AlertConfiguration object, resulting into a list of the
 // prometheus alerts to be created
-func mapAlertsConfiguration(namespace, rateLimitUnit string, rateLimitRequestsPerUnit uint32, requestsAllowedPerSecond float64, alertsConfig map[string]*marin3rconfig.AlertConfig, grafanaConsoleURL string) ([]resources.AlertConfiguration, error) {
+func mapAlertsConfiguration(logger *logrus.Entry, namespace, rateLimitUnit string, rateLimitRequestsPerUnit uint32, requestsAllowedPerSecond float64, alertsConfig map[string]*marin3rconfig.AlertConfig, grafanaDashboardURL string) ([]resources.AlertConfiguration, error) {
 	result := make([]resources.AlertConfiguration, 0, len(alertsConfig))
 
 	for alertName, alertConfig := range alertsConfig {
-		usageFrequencyMins, err := intervalToMinutes(alertConfig.Period)
-		if err != nil {
-			return nil, err
+		switch alertConfig.Type {
+		case marin3rconfig.AlertTypeSpike:
+			expr := fmt.Sprintf(
+				"max_over_time((increase(ratelimit_service_rate_limit_apicast_ratelimit_generic_key_slowpath_total_hits[1m]))[%s:]) /2 > %d",
+				alertConfig.Period, rateLimitRequestsPerUnit)
+			annotations := map[string]string{
+				"message":        fmt.Sprintf("hard limit of %d breached at least once in the last %s", rateLimitRequestsPerUnit, alertConfig.Period),
+				"grafanaConsole": grafanaDashboardURL,
+			}
+			alert := mapSpikeAlert(alertConfig, alertName, namespace, expr, annotations)
+			result = append(result, alert)
+		case marin3rconfig.AlertTypeThreshold:
+
+			usageFrequencyMins, err := intervalToMinutes(alertConfig.Period)
+			if err != nil {
+				return nil, err
+			}
+			requestsAllowedOverTimePeriod := requestsAllowedPerSecond * float64(usageFrequencyMins*60)
+
+			minRateValue, maxRateValue, err := parsePercenteageRange(
+				alertConfig.Threshold.MinRate,
+				alertConfig.Threshold.MaxRate,
+			)
+			if err != nil {
+				return nil, err
+			}
+
+			lowerExpr := increaseExpr(totalRequestsMetric, alertConfig.Period, ">=", requestsAllowedOverTimePeriod, &minRateValue)
+			upperExpr := increaseExpr(totalRequestsMetric, alertConfig.Period, "<=", requestsAllowedOverTimePeriod, maxRateValue)
+
+			// Get the complete expression by ANDing the lower and the upper if the
+			// upper limit is set, if not, assign the lower one
+			expr := *lowerExpr
+			upperMessage := "*"
+			if upperExpr != nil {
+				expr = fmt.Sprintf("%s and %s", expr, *upperExpr)
+				upperMessage = *alertConfig.Threshold.MaxRate
+			}
+			annotations := map[string]string{
+				"message": fmt.Sprintf(
+					"Total API usage in your API Management service is between %s%% and %s%% of the allowable threshold, %d requests per %s, during the last %s",
```

Comment:
Message is showing two `%` symbols:

![Screenshot 2020-11-19 at 15 55 25](https://user-images.githubusercontent.com/1999769/99690156-bc1a8580-2a7f-11eb-9f24-18e724f00142.png)